atom/renderer/api/lib/desktop-capturer.coffee

```diff
@@ -0,0 +1,10 @@
+ipc = require 'ipc'
+NativeImage = require 'native-image'
+
+getSources = (options, callback) ->
+  ipc.send 'ATOM_BROWSER_DESKTOP_CAPTURER_GET_SOURCES', options
+  ipc.once 'ATOM_RENDERER_DESKTOP_CAPTURER_RESULT', (sources) ->
```

Comment:
> Node will show the warning, you can try:
> for (var i = 0; i < 20; ++i) process.once('test', function(){});

I also verify it(looks like my previous test code is incorrect). Sorry for the wrong information.

> Also current implement have problems if you call getSources for multiple times at once:
> dc.getSources {}, -> console.log('done')
> dc.getSources {}, -> console.log('done')
> Each callback will be called twice since ATOM_RENDERER_DESKTOP_CAPTURER_RESULT is sent back twice.

Why the callback will be called twice? The callback here will just be called once which is guranteed by the `ipc.once` code. I also test it with the following code, the output is expected.

```
for (var i = 0; i < 10; ++i) {
  dc.getSources({types:['window']}, function() {console.log('done'); });
}
```